feature-libs/checkout/base/core/facade/checkout-delivery-modes.service.spec.ts

```diff
@@ -0,0 +1,325 @@
+          useClass: MockCheckoutDeliveryModesConnector,
+        },
+        { provide: CheckoutQueryFacade, useClass: MockCheckoutQueryFacade },
+      ],
+    });
+
+    service = TestBed.inject(CheckoutDeliveryModesService);
+    connector = TestBed.inject(CheckoutDeliveryModesConnector);
+    store = TestBed.inject(MockStore);
+    checkoutQuery = TestBed.inject(CheckoutQueryFacade);
+    eventService = TestBed.inject(EventService);
+  });
+
+  it(`should inject CheckoutDeliveryModesService`, inject(
+    [CheckoutDeliveryModesService],
+    (checkoutDeliveryModesService: CheckoutDeliveryModesService) => {
+      expect(checkoutDeliveryModesService).toBeTruthy();
+    }
+  ));
+
+  describe(`getSupportedDeliveryModesState`, () => {
+    it(`should call the checkoutDeliveryModesConnector.getSupportedModes()`, (done) => {
+      spyOn(connector, 'getSupportedModes').and.callThrough();
+
+      service
+        .getSupportedDeliveryModesState()
+        .pipe(take(1))
+        .subscribe((state) => {
+          expect(state).toEqual({
+            loading: false,
+            error: false,
+            data: mockSupportedDeliveryModes,
+          });
+          expect(connector.getSupportedModes).toHaveBeenCalledWith(
+            mockUserId,
+            mockCartId
+          );
+          done();
+        });
+    });
+
+    it(`should unsuccessfully backOff on Jalo error and put the error in the state`, fakeAsync(() => {
+      spyOn(connector, 'getSupportedModes').and.returnValue(
+        throwError(mockJaloError)
+      );
+
+      let resultState: QueryState<DeliveryMode[] | undefined> | undefined;
+      const subscription = service
+        .getSupportedDeliveryModesState()
+        .subscribe((result) => (resultState = result));
+
+      tick(4200);
+
+      expect(resultState).toEqual({
+        loading: false,
+        error: <Error>mockJaloError,
+        data: undefined,
+      });
+      subscription.unsubscribe();
+    }));
+
+    it(`should successfully backOff on Jalo error and recover after the 2nd attempt`, fakeAsync(() => {
+      spyOn(connector, 'getSupportedModes').and.returnValues(
+        // first attempt
+        throwError(mockJaloError),
+        // second attempt
+        throwError(mockJaloError),
+        // third time the charm
+        of(mockSupportedDeliveryModes)
+      );
+
+      let resultState: QueryState<DeliveryMode[] | undefined> | undefined;
+      const subscription = service
+        .getSupportedDeliveryModesState()
+        .subscribe((result) => (resultState = result));
+
+      // 1*1*300 = 300
+      tick(300);
+      expect(resultState).toEqual({
+        loading: true,
+        error: false,
+        data: undefined,
+      });
+
+      // 2*2*300 = 1200
+      tick(1200);
+      expect(resultState).toEqual({
+        loading: false,
+        error: false,
+        data: mockSupportedDeliveryModes,
+      });
+      subscription.unsubscribe();
+    }));
+  });
+
```

Comment:
Add test for describe('getSupportedDeliveryModes', () =>

Reply:
I knew I should've added the method :P
jk, added the missing test, thanks :)

Reply:
nana hhaha I saw you added it for the payment one or was it address, but didn't do it for this so I had to say it :P